jax/numpy/lax_numpy.py

```diff
@@ -100,6 +100,10 @@ def __instancecheck__(self, instance):
       return isinstance(instance, _arraylike_types)
 
 class ndarray(onp.ndarray, metaclass=_ArrayMeta):
+  dtype: onp.dtype
+  shape: Sequence[int]
```

Comment:
Wouldn't Tuple[int, ...] be more accurate?

Reply:
Even better, let's define `core.Shape` and we use it everywhere.

Reply:
> Wouldn't Tuple[int, ...] be more accurate?

I was being defensive because I'm not sure whether we can reasonably expect that shapes are *always* tuples -- is there existing code that assumes they are lists?

> Even better, let's define `core.Shape` and we use it everywhere.

Agreed. Will do that now.

Reply:
> Even better, let's define `core.Shape` and we use it everywhere.

Actually, now that I'm looking at it, core.py is already pretty crowded and does lots of different things. Maybe it's worth splitting type definitions/aliases like these into a separate file like `types.py`. WDYT?

Reply:
Have changed this to Tuple[int, ...] for now